libfwupdplugin/fu-firmware.h

```diff
@@ -122,6 +123,14 @@ struct _FuFirmwareClass {
  * Since: 1.8.2
  **/
 #define FU_FIRMWARE_FLAG_HAS_STORED_SIZE (1u << 5)
+/**
+ * FU_FIRMWARE_FLAG_SEARCH_FOR_MAGIC:
+ *
+ * Search for the magic anywhere in the firmware image.
+ *
+ * Since: 1.8.3
+ **/
+#define FU_FIRMWARE_FLAG_SEARCH_FOR_MAGIC (1u << 6)
```

Comment:
I think about the fact that some firmware can be GB.  I wonder if you want to have flags to separately search header / footer / middle.

Reply:
I think this kind of searching only makes sense for small firmware. I've added a check for < 64MB and also fixed some of the failure messages to be more helpful.